### DISCUSSION TITLE
Can add an ARM64EC build?

### DISCUSSION THREAD

**Discussion by kimjw0820** [microsoft/Detours#304] [Q&A]
Can add an ARM64EC build?

The ARM64EC solution platform was added in visual studio 2022, but the actual build is done in ARM64.
Additionally, even if set to ARM64EC, _ARM64_ from the list below will be activated.

#if !defined(_68K_) && !defined(_MPPC_) && !defined(_X86_) && !defined(_IA64_) && !defined(_AMD64_) && !defined(_ARM_) && !defined(_ARM64_) && !defined(_ARM64EC_) && defined(_M_ARM64)
#define _ARM64_
#endif

#if !defined(_68K_) && !defined(_MPPC_) && !defined(_X86_) && !defined(_IA64_) && !defined(_ARM_) && !defined(_ARM64_) && !defined(_ARM64EC_) && defined(_M_ARM64EC)
#define _ARM64EC_
#endif

detours is built with MakeFile so I don't know what information to add.

I asked a question about ARM64 below, but I am not interested, so I am researching other methods.
https://github.com/microsoft/Detours/issues/292

**Comment by kimjw0820** [comment]
I succeeded in building ARM64EC by changing the build format from MakeFile to static library.

**Reply by Ritchie1108** [reply]
hey bro, how did you do it?

**Reply by shijunz** [reply]
same questions? how to build it and then verify it ok?
